### PR TITLE
Feature/wmt 175/court reports

### PR DESCRIPTION
**Pull request by tsheppard01**
Adds invdividual/team etc court-reporter-overview DB views.  Also adds court_reports_workload/workload_point/workload_points_calculation DB tables.

### CODE REVIEW

File: test/integration/services/data/test-get-workload-points.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('services/data/get-workload-points', function () {
       })
   })
 
-  xit('retrieves the latest points configuration', function (done) {
+  it('retrieves the latest points configuration', function (done) {
```
Comment (andrew-js-wright) [inline test/integration/services/data/test-get-workload-points.js]: 👍

File: migrations/staging/20171009120412_stg_add_oral_reports_to_court_reports.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  return knex.schema.withSchema('staging').table('court_reports', function (table) {
+    table.string('oral_reports')
+  }).catch(function (error) {
+    console.log(error)
+    throw error
+  })
+}
+
+exports.down = function (knex, Promise) {
+  return knex.schema.withSchema('staging').table('arms', function (table) {
```
Comment (caoimheb) [inline migrations/staging/20171009120412_stg_add_oral_reports_to_court_reports.js]: Table should be `court_reports`

File: seed/data/dev/00650_workload_owner.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+var tableName = 'workload_owner'
+
+exports.seed = function (knex, Promise) {
```
Comment (caoimheb) [inline seed/data/dev/00650_workload_owner.js]: Why not build this in to the other workload_owner seed file?

File: seed/data/dev/01210_reductions.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+const reductionStatus = require('../../../app/constants/reduction-status')
+
+var tableName = 'reductions'
+var workloadOwnerId
+
+exports.seed = function (knex, Promise) {
```
Comment (caoimheb) [inline seed/data/dev/01210_reductions.js]: Again, why new file?

File: test/integration/services/data/test-get-workload-points.js (hunk 1 above)
Comment (caoimheb) [inline test/integration/services/data/test-get-workload-points.js]: Was the test data changed to resolve this?

File: seed/views/00602_ldu_court_reporter_overview.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+exports.seed = function (knex, promise) {
+  var sql = `CREATE VIEW app.ldu_court_reporter_overview
+    WITH SCHEMABINDING
+    AS
+    SELECT
```
Comment (caoimheb) [inline seed/views/00602_ldu_court_reporter_overview.js]: This file and region outside the scope of this story?

File: seed/views/00601_team_court_reporter_overview.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        MAX(CONCAT(om.forename, ' ', om.surname)) AS name
+      , MAX(iv.grade_code) AS grade_code
+      , MAX(iv.contracted_hours) AS contracted_hours
+      , MAX(iv.reduction_hours) AS reduction_hours
+      , MAX(iv.link_id) AS id
+      , MAX(iv.id) AS link_id
+      , SUM(iv.total_cases_sdrs) AS total_cases_sdrs
+      , SUM(iv.total_cases_fdrs) AS total_cases_fdrs
+      , SUM(iv.total_cases_oral_reports) AS total_cases_oral_reports
+    FROM app.individual_court_reporter_overview iv
```
Comment (caoimheb) [inline seed/views/00601_team_court_reporter_overview.js]: These views will need to be refactored to not reference each other to work with the new indexes

File: migrations/app/20171009124100_app_court_report_workload.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+
```
[the remark this answers is not in the document]
Comment (tsheppard01) [reply migrations/app/20171009124100_app_court_report_workload.js]: changed to reports

File: migrations/app/20171009124100_app_court_report_workload.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+exports.up = function (knex, Promise) {
+  return knex.schema.createTable('court_reports_workload', function (table) {
+    table.increments('id')
+    table.integer('workload_owner_id').unsigned().references('workload_owner.id')
+    table.integer('total_cases_sdr').unsigned().notNullable()
```
[the remark this answers is not in the document]
Comment (tsheppard01) [reply migrations/app/20171009124100_app_court_report_workload.js]: Good point, changed to remove cases

File: migrations/app/20171010090000_court_reports_workload_points.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+
+exports.up = function (knex, Promise) {
```
[the remark this answers is not in the document]
Comment (tsheppard01) [reply migrations/app/20171010090000_court_reports_workload_points.js]: Renaming tables to: court_reports_wokload => court_reports, court_reports_workload_points_calculation => court_reports_calculations,  move court_reports_workload_points => removed, move default_contracted_hours to workload_points

File: migrations/app/20171010091000_court_report_workload_points_calculation.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    table.decimal('reduction_hours').defaultTo(0).notNullable()
+    table.decimal('contracted_hours').notNullable()
+  }).catch(function (error) {
+    console.log(error)
+    throw error
+  })
+}
+
+exports.down = function (knex, Promise) {
+  return knex.schema.dropTable('court_reports_workload_points_calculation')
```
[the remark this answers is not in the document]
Comment (tsheppard01) [reply migrations/app/20171010091000_court_report_workload_points_calculation.js]: done

File: migrations/staging/20171009120412_stg_add_oral_reports_to_court_reports.js (hunk 2 above)
Comment (tsheppard01) [reply migrations/staging/20171009120412_stg_add_oral_reports_to_court_reports.js]: very good spot

File: migrations/app/20171010091000_court_report_workload_points_calculation.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+
```
[the remark this answers is not in the document]
Comment (tsheppard01) [reply migrations/app/20171010091000_court_report_workload_points_calculation.js]: see above comment

File: seed/data/dev/00650_workload_owner.js (hunk 3 above)
Comment (tsheppard01) [reply seed/data/dev/00650_workload_owner.js]: was easier to write it in a new file, but I've now moved it into the original

File: seed/data/dev/01210_reductions.js (hunk 4 above)
Comment (tsheppard01) [reply seed/data/dev/01210_reductions.js]: moved to the other reductions file

File: test/integration/services/data/test-get-workload-points.js (hunk 1 above)
Comment (tsheppard01) [reply test/integration/services/data/test-get-workload-points.js]: Not sure what you mean.  Was this test commented out for a reason?  It passes locally

File: seed/views/00602_ldu_court_reporter_overview.js (hunk 5 above)
Comment (tsheppard01) [reply seed/views/00602_ldu_court_reporter_overview.js]: Needed a way to navigate down to the team level, easiest way was to add the views for national and regional and ldu (they are very similar to the team view)

**Pull request merged**